apps/web/components/posts/create.tsx

```diff
@@ -0,0 +1,47 @@
+import { useMutation } from "@apollo/client";
+import { Dispatch, useState, SetStateAction } from "react";
+import { GET_POSTS, CREATE_POST } from "web/components/queries"; // to obtain the create_post and get_post function
+import { CreatePostNew } from "../../interface/post"
+
+interface PostProps {   
+    setDisplayForm(): void
+}
+
+const CreatePostForm = ({ setDisplayForm }: PostProps) => {
+    const [newContent, setNewContent] = useState ('');
+    const [createPost, {data, loading, error}] = useMutation<CreatePostNew>(CREATE_POST, {refetchQueries: [GET_POSTS]})
```

Comment:
It would be fine to reuse the `Post` interface here:
```suggestion
    const [createPost, {data, loading, error}] = useMutation<Post>(CREATE_POST, {refetchQueries: [GET_POSTS]})
```